src/main/java/bio/terra/service/tabulardata/azure/StorageTableDao.java

```diff
@@ -0,0 +1,192 @@
+package bio.terra.service.tabulardata.azure;
+
+import bio.terra.model.BulkLoadFileState;
+import bio.terra.model.BulkLoadHistoryModel;
+import bio.terra.service.snapshot.exception.CorruptMetadataException;
+import bio.terra.service.tabulardata.LoadHistoryUtil;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StorageTableDao {
+
+  private static final String LOAD_HISTORY_TABLE_NAME_SUFFIX = "LoadHistory";
+
+  /**
+   * Store the results of a bulk file load in an Azure Storage Table
+   *
+   * <p>The table name will be the result of the dataset id passed through {@link
+   * StorageTableDao#toStorageTableNameFromUUID} Entities will be partitioned on the loadTag and
+   * their row keys will be the value of {@link BulkLoadHistoryModel#getFileId()}
+   *
+   * @param serviceClient A service client for the dataset
+   * @param datasetId the id of the dataset
+   * @param loadTag Load tag to partition on
+   * @param loadTime The time the load occurred
+   * @param loadHistoryArray The models to store
+   */
+  public void loadHistoryToAStorageTable(
+      TableServiceClient serviceClient,
+      UUID datasetId,
+      String loadTag,
+      Instant loadTime,
+      List<BulkLoadHistoryModel> loadHistoryArray) {
+    if (loadHistoryArray.isEmpty()) {
+      return;
+    }
+    var tableName = toStorageTableNameFromUUID(datasetId);
+    TableClient client = serviceClient.createTableIfNotExists(tableName);
+    // if the table already exists, the returned client is null and we have to get it explicitly
+    if (client == null) {
+      client = serviceClient.getTableClient(tableName);
+    }
+
+    var base64LoadTag =
+        Base64.getEncoder().encodeToString(loadTag.getBytes(StandardCharsets.UTF_8));
+
+    ListEntitiesOptions options =
+        new ListEntitiesOptions()
```

Comment:
To be safe, do we want to add something like 
```suggestion
        new ListEntitiesOptions()
          .setTop(2)
          ...
        
```